Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -792,523 +474,160 @@ WarpX::ComputeOneWayExtensions() {
-
-                    while(denom >= Sx_ext && neg_face && denom > 0){
-                        neg_face = false;
-                        for (int j_n = -1; j_n < 2; j_n++) {
-                            for (int k_n = -1; k_n < 2; k_n++) {
-                                if(local_avail(j_n + 1, k_n + 1)){
-                                    amrex::Real patch = Sx_ext * Sx(i, j + j_n, k+ k_n) / denom;
-                                    if(Sx_mod(i, j + j_n, k + k_n) - patch <= 0) {
-                                        neg_face = true;
-                                        local_avail(j_n + 1, k_n + 1) = false;
-                                    }
-                                }
-                            }
-                        }
-
-                        denom = local_avail(0, 1) * Sx(i, j - 1, k) +
-                                local_avail(2, 1) * Sx(i, j + 1, k) +
-                                local_avail(1, 0) * Sx(i, j, k - 1) +
-                                local_avail(1, 2) * Sx(i, j, k + 1) +
-                                local_avail(0, 0) * Sx(i, j - 1, k - 1) +
-                                local_avail(2, 0) * Sx(i, j + 1, k - 1) +
-                                local_avail(0, 2) * Sx(i, j - 1, k + 1) +
-                                local_avail(2, 2) * Sx(i, j + 1, k + 1);
-                    }
-
-                    if(denom >= Sx_ext){
-                        Sx_mod(i, j, k) = Sx(i, j, k);
-                        int count = 0;
-                        for (int j_n = -1; j_n < 2; j_n++) {
-                            for (int k_n = -1; k_n < 2; k_n++) {
-                                if(local_avail(j_n + 1, k_n + 1)){
-                                    amrex::Real patch = Sx_ext * Sx(i, j + j_n, k + k_n) / denom;
-                                    borrowing_x_inds[ps + count] = ps + count;
-                                    // Store the information about the intruded face in the dataset of the
-                                    // faces which are borrowing area
-                                    FaceInfoBox::addConnectedNeighbor(j_n, k_n, ps + count,
-                                                                      borrowing_x_neigh_faces);
-                                    borrowing_x_area[ps + count] = patch;
-
-                                    flag_info_face_x(i, j + j_n, k + k_n) = 2;
-                                    Sx_mod(i, j, k) += patch;
-                                    Sx_mod(i, j + j_n, k + k_n) -= patch;
-                                    count+=1;
-                                }
-                            }
-                        }
-                        flag_ext_face_x(i, j, k) = false;
-                    }
-                }
-            },
-            amrex::Scan::Type::exclusive);
 
-    }
+    const amrex::Real dx = cell_size[0];
+    const amrex::Real dy = cell_size[1];
+    const amrex::Real dz = cell_size[2];
```

Comment:
As above: extra paths for 1D/2D/RZ here?

Reply:
Sam answer as above :)